### PR TITLE
[Unix] Create a framework for certbot integration tests: PART 3a

### PR DESCRIPTION
**Pull request by adferrand**
Following #6821, this PR continues to convert certbot integration tests into certbot-ci.

This PR add tests covering on L185-222 in `tests/certbot-boulder-integration.sh`.

### CODE REVIEW

File: certbot-ci/certbot_integration_tests/certbot_tests/test_main.py [hunk 1]
```diff
@@ -23,6 +24,72 @@ def context(request):
...
+        context.certbot([
+            '--domains', certname, '--preferred-challenges', 'http-01', 'run',
+            '--cert-name', certname,
+            '--pre-hook', 'echo wtf.pre >> "{0}"'.format(context.hook_probe),
+            '--post-hook', 'echo wtf.post >> "{0}"'.format(context.hook_probe),
+            '--deploy-hook', 'echo deploy >> "{0}"'.format(context.hook_probe)
+        ])
+
+    assert_hook_execution(context.hook_probe, 'deploy')
+    assert_save_renew_hook(context.config_dir, certname)
```
Comment (bmw) [inline certbot-ci/certbot_integration_tests/certbot_tests/test_main.py]: I suspect we just want to change the code in `assert_save_renew_hook`, but this isn't testing the same thing as [CheckSavedRenewHook](https://github.com/certbot/certbot/blob/d9880721b31ab4e9e9d2fdc38d83ff2bd9078378/tests/certbot-boulder-integration.sh#L142). The shell code is trying to check that there is an entry for `renew_hook` saved into the renewal configuration file.

Comment (adferrand) [reply certbot-ci/certbot_integration_tests/certbot_tests/test_main.py]: Indeed, I did not interpretate this assertion correctly. This is fixed.

File: certbot-ci/certbot_integration_tests/certbot_tests/test_main.py [hunk 2]
```diff
@@ -120,12 +121,12 @@ def test_manual_dns_auth(context):
         '--manual-cleanup-hook', context.manual_dns_cleanup_hook,
         '--pre-hook', 'echo wtf.pre >> "{0}"'.format(context.hook_probe),
         '--post-hook', 'echo wtf.post >> "{0}"'.format(context.hook_probe),
-        '--renew-hook', 'echo renew >> "{0}"'.format(context.hook_probe)
+        '--renew-hook', 'echo deploy >> "{0}"'.format(context.hook_probe)
```
Comment (bmw) [inline certbot-ci/certbot_integration_tests/certbot_tests/test_main.py]: Can we leave this at "renew"?

If for some reason we want to change this to "deploy", I think we should also change the `assert_hook_execution` call down below to "deploy" as well.

Comment (adferrand) [reply certbot-ci/certbot_integration_tests/certbot_tests/test_main.py]: Original tests were using the `deploy` keyword. But it is just a naming convention, without functional impact. I change the word, it makes more sense

**Pull request merged**